modules/fscloud/main.tf

```diff
@@ -94,7 +99,7 @@ module "instance_cbr_rules" {
       },
       {
         name     = "serviceInstance"
-        value    = module.cos_instance.cos_instance_guid
+        value    = module.cos_instance[0].cos_instance_guid
```

Comment:
Similarly, use local.cos_instance_id created above so that CBR references existing COS when appropriate.